[Jules]: Fix CI integration issues

### Agent Prompt

You are a Senior Home Assistant Core Developer and Test Automation Expert. Our automated integration reset script is failing during the 'user' step of the config flow with a HTTP 400 Bad Request.

<detected_errors>
23:15:48 [INFO] POST http://192.168.5.14:8123/api/config/config_entries/flow/01KKD0HYQMVPFZ6BVS2C9G7A3J for step user with payload: {'api_key': '9f81588a4e27a95a4c82746eebc7c9e13e60d6c8', 'organization_id': '1254890'}
23:15:48 [ERROR] Failed to submit step user: 400
</detected_errors>

<instructions>
1. Think step-by-step. Outline why the Home Assistant `/api/config/config_entries/flow/{flow_id}` endpoint returns a 400 status inside <thinking> tags. 
2. Action 1: Check the requirements to prevent regressions and update the documentation to reflect any new testing schema requirements.
3. Action 2: Open `custom_components/meraki_ha/config_flow.py`.
4. Action 3: Inspect the `voluptuous` data schema inside `async_step_user`. Compare it directly against the script's payload: `{'api_key': '...', 'organization_id': '...'}`. 
5. Action 4: Identify any missing required fields (e.g., `name`), type mismatches (e.g., string vs int), or schema defaults that the automated script is failing to satisfy.
6. Action 5: If the schema matches perfectly, inspect the `async_step_user` try/except block. Ensure that exceptions like `ApiClientCommunicationError` or `vol.Invalid` are caught and mapped to the `errors` dictionary rather than bubbling up and crashing the request.
7. Action 6: Provide the specific fix for either the Python test script payload OR the `config_flow.py` schema.
8. Always output the complete merged files and proactively apply optimizations.
</instructions>

<examples>
Example: Schema Mismatch
Positive Alternative: Ensure the test script payload perfectly aligns with the voluptuous schema.
```python
# If config_flow.py expects:
# vol.Schema({vol.Required("api_key"): str, vol.Optional("name", default="Meraki"): str})

# The script MUST send all required fields, and occasionally explicit defaults if the API is strict.

--- a/custom_components/meraki_ha/__init__.py
+++ b/custom_components/meraki_ha/__init__.py
@@ -81,6 +81,22 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating from version %s", entry.version)
+
+    if entry.version == 1:
+        new_data = {**entry.data}
+        if "meraki_api_key" in new_data:
+            new_data["api_key"] = new_data.pop("meraki_api_key")
+
+        hass.config_entries.async_update_entry(entry, data=new_data, version=2)
+
+    _LOGGER.info("Migration to version %s successful", entry.version)
+
+    return True
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """
     Set up Meraki from a config entry.

--- a/custom_components/meraki_ha/config_flow.py
+++ b/custom_components/meraki_ha/config_flow.py
@@ -28,13 +28,18 @@ _LOGGER = logging.getLogger(__name__)
 class MerakiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Meraki."""
 
-    VERSION = 1
+    VERSION = 2
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> FlowResult:
         """Handle the initial step."""
         from .core.api import create_api_client
+        from .core.errors import (
+            InvalidOrgID,
+            MerakiAuthenticationError,
+            MerakiConnectionError,
+        )
         from .schemas import STEP_USER_DATA_SCHEMA
 
         errors: dict[str, str] = {}
@@ -48,6 +53,12 @@ class MerakiConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
                 )
                 await client.async_setup()
                 await client.get_organizations()
+            except MerakiAuthenticationError:
+                errors["base"] = "invalid_auth"
+            except MerakiConnectionError:
+                errors["base"] = "cannot_connect"
+            except InvalidOrgID:
+                errors["base"] = "invalid_org_id"
             except Exception:  # pylint: disable=broad-except
                 _LOGGER.exception("Unexpected exception during config flow")
                 errors["base"] = "cannot_connect"

--- a/custom_components/meraki_ha/const/config.py
+++ b/custom_components/meraki_ha/const/config.py
@@ -7,7 +7,7 @@ from typing import Final
 CONF_INTEGRATION_TITLE: Final = "Meraki"
 """Title for the integration."""
 
-CONF_MERAKI_API_KEY: Final = "meraki_api_key"
+CONF_MERAKI_API_KEY: Final = "api_key"
 """Configuration key for the Meraki API key."""
 
 CONF_MERAKI_ORG_ID: Final = "organization_id"

--- a/custom_components/meraki_ha/core/utils/validation_utils.py
+++ b/custom_components/meraki_ha/core/utils/validation_utils.py
@@ -6,9 +6,10 @@ import re
 
 import homeassistant.helpers.config_validation as cv
 import voluptuous as vol
-from homeassistant.const import CONF_API_KEY, CONF_NAME
+from homeassistant.const import CONF_NAME
 
 from custom_components.meraki_ha.const.config import (
+    CONF_MERAKI_API_KEY,
     CONF_MERAKI_ORG_ID,
     CONF_SCAN_INTERVAL,
     DEFAULT_SCAN_INTERVAL,
@@ -65,7 +66,7 @@ def validate_org_id(value: str) -> str:
 # Main config schema
 CONFIG_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_API_KEY): validate_api_key,
+        vol.Required(CONF_MERAKI_API_KEY): validate_api_key,
         vol.Required(CONF_MERAKI_ORG_ID): validate_org_id,
         vol.Optional(CONF_NAME): cv.string,
         vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): vol.All(

--- a/docs/requirements/requirements.md
+++ b/docs/requirements/requirements.md
@@ -75,4 +75,7 @@ This document verifies the state of the codebase against the requirements for th
   - **[VERIFIED]** Gateway, Switch, and AP pills now support clickable actions, defaulting to `more-info`.
   - **[VERIFIED]** Added configuration fields to the card editor for managing these tap actions.
 
-This verification confirms the need for the planned refactoring steps. The new requirements (R4, R5, R6, R7, R8, R9, R10, R11, R12, R13, R14, R15, R16, R17) are now considered part of the standard for this integration.
+- **R18: Schema Consistency:**
+  - **[IN PROGRESS]** Implementation of configuration schema consistency to ensure alignment between constants, translation files, and automated testing scripts. This includes migrating legacy keys (e.g., `meraki_api_key`) to standardized keys (e.g., `api_key`).
+
+This verification confirms the need for the planned refactoring steps. The new requirements (R4, R5, R6, R7, R8, R9, R10, R11, R12, R13, R14, R15, R16, R17, R18) are now considered part of the standard for this integration.

--- a/tests/test_migration.py
+++ b/tests/test_migration.py
@@ -1,0 +1,22 @@
+"""Test migration for Meraki."""
+from homeassistant.core import HomeAssistant
+from custom_components.meraki_ha import async_migrate_entry
+from pytest_homeassistant_custom_component.common import MockConfigEntry
+
+async def test_migration_v1_to_v2(hass: HomeAssistant):
+    """Test migrating a config entry from version 1 to 2."""
+    entry = MockConfigEntry(
+        domain="meraki_ha",
+        version=1,
+        data={
+            "meraki_api_key": "test-key",
+            "organization_id": "123456",
+        },
+    )
+    entry.add_to_hass(hass)
+
+    assert await async_migrate_entry(hass, entry)
+
+    assert entry.version == 2
+    assert entry.data["api_key"] == "test-key"
+    assert "meraki_api_key" not in entry.data

